core/src/main/java/net/opentsdb/storage/MockDataStore.java

```diff
@@ -80,19 +78,20 @@
   
   @SuppressWarnings("unchecked")
   @Override
-  public Deferred<Object> write(final TimeSeriesValue<?> value, 
+  public Deferred<Object> write(final TimeSeriesId id,
+                                final TimeSeriesValue<?> value, 
                                 final TsdbTrace trace,
                                 final Span upstream_span) {
-    Map<TypeToken<?>, MockSpan<?>> type = database.get(value.id());
+    Map<TypeToken<?>, MockSpan<?>> type = database.get(id);
     if (type == null) {
       type = Maps.newHashMap();
-      database.put(value.id(), type);
+      database.put(id, type);
     }
     
     MockSpan<?> span = type.get(value.type());
     if (span == null) {
       if (value.type() == NumericType.TYPE) {
-        span = new MockSpan<NumericType>();
+        span = new MockSpan<NumericType>(id);
```

Comment:
Pass in an Optional<Span>?

Reply:
ID moved to a higher level to avoid duplication so it's no longer used here.